### PR TITLE
[#1074]: Align names of protocol adapter main application classes

### PR DESCRIPTION
**Pull request by ctron**
This is proposal to fix issue #1074 … by aligning all protocol adapter main application classes to the pattern of `org.eclipse.hono.adapter.<foo>.impl.Application`.

### CODE REVIEW

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/Application.java [hunk 1]
```diff
@@ -21,7 +21,9 @@
 /**
  * The Hono AMQP main application class.
  */
-@ComponentScan(basePackages = { "org.eclipse.hono.adapter.amqp", "org.eclipse.hono.service.metric" })
+@ComponentScan("org.eclipse.hono.adapter.amqp")
+@ComponentScan("org.eclipse.hono.adapter.amqp.impl")
+@ComponentScan("org.eclipse.hono.service.metric")
```
Comment (sophokles73) [inline adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/Application.java]: I am not quite convinced that this is the same as specifying the packages in a single annotation. Do you have any source where the semantics of multiple instances of the same annotation on a type is defined, in particular in the Spring context?

Comment (ctron) [reply adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/Application.java]: I think it is:

* https://docs.oracle.com/javase/tutorial/java/annotations/repeating.html
* https://github.com/spring-projects/spring-framework/blob/bc283ed7d70570bffba0f2a8c82b8e666a87d9f4/spring-context/src/main/java/org/springframework/context/annotation/ComponentScan.java#L58
* https://github.com/spring-projects/spring-framework/blob/4.3.x/spring-context/src/main/java/org/springframework/context/annotation/ComponentScans.java

File: adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/Application.java [hunk 2]
```diff
@@ -22,7 +22,9 @@
 /**
  * The Hono MQTT adapter main application class.
  */
-@ComponentScan(basePackages = { "org.eclipse.hono.adapter.mqtt", "org.eclipse.hono.service.metric" })
+@ComponentScan("org.eclipse.hono.adapter.mqtt")
+@ComponentScan("org.eclipse.hono.adapter.mqtt.impl")
```
Comment (sophokles73) [inline adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/Application.java]: This should already be covered by *base package* `org.eclipse.hono.adapter.mqtt`, shouldn't it?

Comment (ctron) [reply adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/Application.java]: I wasn't able to find any explicit documentation, but running a test confirmed that this does work. So I dropped the annotations for "impl".

**Comment by ctron** [comment]
@sophokles73 Is it ok to merge, from your perspective?

**Pull request merged**